core/templates/dev/head/pages/exploration_editor/statistics_tab/LearnerActionRenderServiceSpec.js

```diff
@@ -0,0 +1,316 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the learner action render service.
+ */
+
+describe('Learner Action Render Service', function() {
+  beforeEach(module('oppia'));
+
+  describe('Test learner action render service functions', function() {
+    beforeEach(inject(function($injector) {
+      this.laof = $injector.get('LearnerActionObjectFactory');
+      this.ps = $injector.get('PlaythroughService');
+      spyOn(this.ps, 'isPlayerExcludedFromSamplePopulation').and.returnValue(
+        false);
+      spyOn(this.ps, 'isExplorationWhitelisted').and.returnValue(true);
+      this.lars = $injector.get('LearnerActionRenderService');
+      this.ps.initSession('expId1', 1);
+    }));
+
+    it('should split up EarlyQuit learner actions into display blocks.',
+      function() {
+        this.ps.recordExplorationStartAction('stateName1');
+        this.ps.recordAnswerSubmitAction(
+          'stateName1', 'stateName2', 'Continue', '', 'Welcome', 30);
+        this.ps.recordAnswerSubmitAction(
+          'stateName2', 'stateName2', 'TextInput', 'Hello', 'Try again', 30);
+        this.ps.recordExplorationQuitAction('stateName2', 120);
+
+        var learnerActions = this.ps.getPlaythrough().actions;
+
+        var displayBlocks = this.lars.getDisplayBlocks(learnerActions);
+        expect(displayBlocks.length).toEqual(1);
```

Comment:
These expect blocks can be condensed into a single one:

```javascript
expect(displayBlocks).toEqual([
  this.laof.createNew(...),
  jasmine.any(Object),
  jasmine.any(Object),
  this.laof.createNew(...)
]);
```

If possible, please also make a more descriptive matcher for the objects in indicies 1 and 2.

Reply:
Done.